crates/shielded-pool/src/component/note_manager.rs

```diff
@@ -16,6 +12,42 @@ use crate::{event, state_key};
 
 use super::SupplyWrite;
 
+#[derive(Clone)]
+pub enum StatePayload {
+    RolledUp(note::Commitment),
+    Note {
+        source: NoteSource,
+        note: Box<NotePayload>,
+    },
+}
```

Comment:
This `StatePayload` is only a subset of the possible state payload variants, meaning that we'll need to have another `StatePayload` somewhere else with more variants, right?

I'm wondering if it would be simpler to not have this enum at all, and instead have separate handling for note payloads and bare commitments -- what's the situation where we want to be handling "either bare commitment OR note payload" that's not "either bare commitment OR note payload OR swap payload"?